### ISSUE TITLE
Unicode object keys causing ESLint error: `TypeError: Graphemer is not a constructor`

### ISSUE THREAD

**Issue by Krutoy242** [eslint-stylistic/eslint-stylistic#216]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Using unicode as object keys fails linter and make ESLint unable to work with file.

ESLint config have this line:
```js
'style/key-spacing' : ['error', { align: 'colon' }]
```

Minimal code:
```ts
const list = {
	'♥': 1,
	'♦': 2,
};
```

Result:
```log
[Error - 4:24:29 PM] An unexpected error occurred:
[Error - 4:24:29 PM] TypeError: Graphemer is not a constructor
Occurred while linting [...]\scripts\category\creative.ts:3
Rule: "style/key-spacing"
    at Object.getGraphemeCount ([...]\node_modules\.pnpm\@stylistic+eslint-plugin-js@1.5.0_eslint@8.55.0\node_modules\@stylistic\eslint-plugin-js\dist\utils.js:337:16)
    at getKeyWidth ([...]\node_modules\.pnpm\@stylistic+eslint-plugin-js@1.5.0_eslint@8.55.0\node_modules\@stylistic\eslint-plugin-js\dist\key-spacing.js:348:20)
```

Line referenced is https://github.com/eslint-stylistic/eslint-stylistic/blob/c1c772ec09ff4133c618a48db44ba03a900e68a7/packages/eslint-plugin-js/utils/string-utils.ts#L21

### Reproduction

https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-26t8nh?file=playground%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
